vendor/zkllvm-metacraft-circuits/src/circuit_utils/ssz_utils.h

```diff
@@ -1,19 +1,22 @@
 #pragma once
 
 #include "circuit_byte_utils.h"
+#include "base_types.h"
 
 using namespace circuit_byte_utils;
 
 namespace ssz_utils {
 
-    template<size_t MERKLE_DEPTH>
-    Bytes32
-        ssz_restore_merkle_root(const Bytes32& leaf, const std::array<Bytes32, MERKLE_DEPTH>& branch, uint64_t gindex) {
+    template<size_t MERKLE_DEPTH, bool F>
+    HashType ssz_restore_merkle_root(const HashType& leaf,
+                                     const static_vector<HashType, MERKLE_DEPTH, F>& branch,
+                                     uint64_t gindex,
+                                     const uint64_t depth = MERKLE_DEPTH) {
```

Comment:
Do you really need the depth parameter?

Reply:
I have removed the depth parameter, the proof paths are always the same length.

Reply:
We can remove the depth parameter. I will have to extract the `fill_zero` logic out of the circuit code in order for it to work. will do it in a separate PR.

Reply:
It is not needed. Also the logic to fill in zero hashes for different levels is extracted out of the circuit.